File: Guests/Python3/sanity/TestFuncs.py
import time

def hello_world():
    print('Hello World, from Python3')


def returns_an_error():
    raise Exception('Error')


def div_integers(x, y):
    return x/ y


def join_strings(arr):
	res = ','.join(arr)
	return res

five_seconds = 5
def wait_a_bit(secs):
	time.sleep(secs)
	return None

class testmap:
	name: str
	curdict: dict

	def __init__(self):
		self.curdict = dict()

	def set(self, k, v):
		self.curdict[k] = v

	def get(self, k):
		return self.curdict[k]

	def contains(self, k):
		return self.curdict[k] is not None

File: Guests/Go/sanity/TestFuncs.go
package TestFuncs

import (
	"strings"
)

func HelloWorld() {
	println("Hello World, From Go!")
}

func ReturnsAnError(){
	panic("An error from ReturnsAnError")
}

func DivIntegers(x int, y int) float32{

	if y == 0{
		panic("Divisor is 0")
	}

	return float32(x) / float32(y)
}

func JoinStrings(arrs []string) string{
	return strings.Join(arrs, ",")
}

type TestMap struct{
	m map[string]interface{}
}

func NewTestMap() *TestMap{
	return &TestMap{ m: make(map[string]interface{})}
}

func (this *TestMap) SetKey(k string, v interface{}){
	this.m[k] = v
}

func (this *TestMap) GetKey(k string) interface{}{
	return this.m[k]
}

func (this *TestMap) ContainsKey(k string) bool{
	_, found := this.m[k]
	return found
}
